http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala

```diff
@@ -342,6 +343,215 @@ class AKSInterpreter[F[_]](config: AKSInterpreterConfig,
 
     } yield ()
 
+  override def updateAndPollApp(params: UpdateAKSAppParams)(implicit ev: Ask[F, AppContext]): F[Unit] = {
+    for {
+      ctx <- ev.ask
+
+      workspaceId <- F.fromOption(
+        params.workspaceId,
+        AppUpdateException(
+          s"${params.appName} must have a Workspace in the Azure cloud context | trace id: ${ctx.traceId}"
+        )
+      )
+
+      // Grab records from the database
+      dbAppOpt <- KubernetesServiceDbQueries
+        .getActiveFullAppByName(CloudContext.Azure(params.cloudContext), params.appName)
+        .transaction
+      dbApp <- F.fromOption(
+        dbAppOpt,
+        AppNotFoundException(CloudContext.Azure(params.cloudContext),
+                             params.appName,
+                             ctx.traceId,
+                             "No active app found in DB"
+        )
+      )
+      _ <- logger.info(ctx.loggingCtx)(s"Updating app $params.appName in workspace $params.workspaceId")
+
+      app = dbApp.app
+      namespaceName = app.appResources.namespace.name
+      kubernetesNamespace = KubernetesNamespace(namespaceName)
+
+      // Grab the LZ and storage container information associated with the workspace
```

Comment:
The other app messages (create/delete) get the `landingZoneResources` and workspace-storage container from the pubsub message, instead of querying it here. If we're triggering UpdateApp messages via Leo API then maybe we should do it the same way. Not sure I have a strong preference though…

Reply:
Agreed, let me take a stab at the front leo implementation and see if I can harmonize a bit more with what the create/delete endpoints are already doing